samples/verona/lang.cc

```diff
@@ -1337,9 +1479,162 @@ namespace verona
       }};
   }
 
+  PassDef partialapp()
+  {
+    // This should happen after `lambda` (so that anonymous types get partial
+    // application), after `autocreate` (so that constructors get partial
+    // application), and after `defaultargs` (so that default arguments don't
+    // get partial application).
+
+    // This means that partial application can't be written in terms of lambdas,
+    // but instead has to be anonymous classes. There's no need to check for
+    // non-local returns.
+    return {
+      dir::bottomup | dir::once,
+      {T(Function)[Function]
+         << ((T(Ref) / T(DontCare))[Ref] * Name[Id] *
+             T(TypeParams)[TypeParams] * T(Params)[Params]) >>
+       [](Match& _) {
+         auto f = _(Function);
+         auto ref = _(Ref);
+         auto id = _(Id);
+         auto tp = _(TypeParams);
+         auto params = _(Params);
+         size_t start_arity = 0;
+         auto end_arity = params->size();
+         auto parent = f->parent()->parent();
+         auto defs = parent->lookdown(id->location());
+         auto tn = parent->at(wf / Class / Ident, wf / TypeTrait / Ident);
+         Node call = (ref->type() == Ref) ? CallLHS : Call;
+
+         for (auto def : defs)
```

Comment:
Perhaps comment that this is for finding smaller arity version.   That is if you have definitions for `f(a,b,c)` and `f(a,b,c,d,e)` then only `f(a,b,c,d)` is provided by the five argument version, and the 2 and 1 argument version come from `f(a,b,c)`.